sdks/java/io/parquet/src/main/java/org/apache/beam/sdk/io/parquet/ParquetIO.java

```diff
@@ -192,21 +202,40 @@ public void populateDisplayData(DisplayData.Builder builder) {
     @Nullable
     abstract Schema getSchema();
 
+    @Nullable
+    abstract GenericData getAvroDataModel();
+
+    abstract Builder toBuilder();
+
     @AutoValue.Builder
     abstract static class Builder {
       abstract Builder setSchema(Schema schema);
 
+      abstract Builder setAvroDataModel(GenericData model);
+
       abstract ReadFiles build();
     }
 
+    public ReadFiles withAvroDataModel(GenericData model) {
```

Comment:
Could you add Javadoc for this public method and maybe an example into main class?

Reply:
Will do.